Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -133,6 +133,12 @@ using SourceLinkAccessorDelegate =
     Delegate<std::pair<source_link_iterator_t, source_link_iterator_t>(
         const Surface&)>;
 
+/// expected max number of track states that are expected to be added by
+/// stateCandidateCreator
+/// @note if the number of states exceeds this number dynamic memory allocation will occur.
+//.       the number is chosen to yield a container size of 64 bytes.
+static constexpr std::size_t s_maxBranchesPerSurface = 10;
```

Comment:
Can this be moved next to `TrackStateCandidateCreator`?

Reply:
It could but that would couple the return type of the new delegate to this one particular implementation.  I also do not see any other rype where this would fit in a kind of natural way.

Reply:
Hm I am just a bit worried of this entering the `Acts` namespace and then be used by other components. Maybe we can just capture it in a nested CKF like namespace?